src/Evolution/Systems/Cce/Tags.hpp

```diff
@@ -0,0 +1,232 @@
+/// surface.
+template <typename Tag>
+struct BoundaryValue : db::PrefixTag, db::SimpleTag {
+  using type = Scalar<SpinWeighted<ComplexDataVector, Tag::type::type::spin>>;
+  using tag = Tag;
+  static std::string name() noexcept {
+    return "BoundaryValue(" + Tag::name() + ")";
+  }
+};
+
+/// A prefix tag representing the coefficient of a pole part of the right-hand
+/// side of a singular differential equation
+template <typename Tag>
+struct PoleOfIntegrand : db::PrefixTag, db::SimpleTag {
+  using type = Scalar<SpinWeighted<ComplexDataVector, Tag::type::type::spin>>;
+  using tag = Tag;
+  static std::string name() noexcept {
+    return "PoleOfIntegrand(" + Tag::name() + ")";
+  }
+};
+
+/// A prefix tag representing the regular part of the right-hand side of a
+/// regular differential equation
+template <typename Tag>
+struct RegularIntegrand : db::PrefixTag, db::SimpleTag {
+  using type = Scalar<SpinWeighted<ComplexDataVector, Tag::type::type::spin>>;
+  using tag = Tag;
+  static std::string name() noexcept {
+    return "RegularIntegrand(" + Tag::name() + ")";
+  }
+};
+
+/// A prefix tag representing a linear factor that acts on `Tag`. To determine
+/// the spin weight, It is assumed that the linear factor plays the role of
+/// \f$L\f$ in an equation of the form,
+/// \f$ (y - 1) \partial_y H + L H + L^\prime \bar{H} = A + (1 - y) B \f$
+template <typename Tag>
+struct LinearFactor : db::PrefixTag, db::SimpleTag {
+  using type = Scalar<SpinWeighted<ComplexDataVector, 0>>;
+  using tag = Tag;
+  static std::string name() noexcept {
+    return "LinearFactor(" + Tag::name() + ")";
+  }
+};
+
+/// A prefix tag representing a linear factor that acts on `Tag`. To determine
+/// the spin weight, it is assumed that the linear factor plays the role of
+/// \f$L^\prime\f$ in an equation of the form,
+/// \f$ (y - 1) \partial_y H + L H + L^\prime \bar{H} = A + (1 - y) B \f$
+template <typename Tag>
+struct LinearFactorForConjugate : db::PrefixTag, db::SimpleTag {
+  using type =
+      Scalar<SpinWeighted<ComplexDataVector, 2 * Tag::type::type::spin>>;
+  using tag = Tag;
+  static std::string name() noexcept {
+    return "LinearFactorForConjugate(" + Tag::name() + ")";
+  }
+};
+
+// Below are additional values which are frequently used in CCE calculations,
+// and therefore worth caching
+
+/// Coordinate value \f$(1 - y)\f$, which will be cached and sent to the
+/// implementing functions
+struct OneMinusY : db::SimpleTag {
+  using type = Scalar<SpinWeighted<ComplexDataVector, 0>>;
+  static std::string name() noexcept { return "OneMinusY"; }
+};
+
+/// A tag for the first time derivative of the worldtube parameter
+/// \f$\partial_u R\f$.
+struct DuR : db::SimpleTag {
+  using type = Scalar<SpinWeighted<ComplexDataVector, 0>>;
+  static std::string name() noexcept { return "DuR"; }
+};
+
+/// The value \f$\partial_u R / R\f$, where \f$R\f$ is the variable associated
+/// with the shape of the initial worldtube data.
```

Comment:
Could you provide the definition of R also in at least the Dy tag above?  Perhaps also for DuR immediately above.. you do say "worldtube parameter" there, but this phrase seems clearer to me.